client_code/storage.py

```diff
@@ -5,79 +5,97 @@
-
-    def _filter_store(self):
-        return filter(lambda key: key.startswith(_prefix), self._store.keys())
-
-    def _map_store(self, predicate):
-        self._check_store()
-        return map(predicate, self._filter_store())
+        self._driver = driver
+        self._name = name
+
+    def is_available(self):
+        """check if the store object is available and accessible."""
+        # in some browsers localStorageWrapper might not be available
+        if not self._store.supports(self._driver):
+            # we cn't rely on this method - it's just for browser support
+            return False
+        try:
+            self._store.length()
+            # call a method in the store to activate the store
+            return True
+        except Exception:
+            return False
 
     def __getitem__(self, key):
-        ret = self._store.getItem(self._mangle_key(key))
-        if ret is None:
-            raise KeyError(key)
-        return _json.loads(ret)
+        if key in self._store.keys():
+            return _deserialize(self._store.getItem(key))
+        raise KeyError(key)
 
     def __setitem__(self, key, val):
-        key = self._mangle_key(key)
-        try:
-            val = _json.dumps(val)
-        except Exception as e:
-            raise type(e)(f"There was a problem converting the value into json: {e}")
         self._store.setItem(key, val)
 
     def __delitem__(self, key):
-        self._store.removeItem(self._mangle_key(key))
+        # we can't block here so do a Promise hack
+        _window.Promise(lambda res, rej: self._store.removeItem(key))
+        return None
 
     def __contains__(self, key):
-        return self._mangle_key(key) in self._store
+        return key in self._store.keys()
 
     def __repr__(self):
-        pairs = ", ".join(f"{key!r}: {val!r}" for key, val in self.items())
-        return f"Storage({{{pairs}}})"
+        # we can't print the items like a dictionary since we get a SuspensionError here
+        return f"<{driver_names[self._driver]} for {self._name!r} store>"
 
     def __iter__(self):
         return self.keys()
 
+    def __len__(self):
+        return self._store.length()
+
     def keys(self):
         """returns the keys for local storage as an iterator"""
-        return self._map_store(lambda key: key[_prefix_len:])
+        return self._store.keys()
 
     def items(self):
         """returns the items for local storage as an iterator"""
-        return self._map_store(lambda key: (key[_prefix_len:], self.__getitem__(key)))
+        return (
+            (key, _deserialize(self._store.getItem(key))) for key in self._store.keys()
+        )
 
     def values(self):
         """returns the values for local storage as an iterator"""
-        return self._map_store(lambda key: self.__getitem__(key))
+        return (_deserialize(self._store.getItem(key)) for key in self._store.keys())
 
     def put(self, key, value):
```

Comment:
using 'put' slightly implies a stack structure. How about 'store' or 'save'?

Reply:
FWIW - from a noobs perspective I also think that 'store' is less daunting.

Reply:
good idea - `store` it shall be. 
I'll alias `put` with `store` and remove `put` from the api docs.